src/mono/System.Private.CoreLib/src/System/Runtime/DependentHandle.cs

```diff
@@ -0,0 +1,167 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime
+{
+    // The ephemeron type is checked in the Mono runtime in "object.c" as follows:
+    //
+    // m_class_get_image (klass) == mono_defaults.corlib && !strcmp ("Ephemeron", m_class_get_name (klass))
+    //
+    // As such, the namespace it belongs to in the managed runtime doesn't matter: the VM will just check
+    // that the type name matches, and that the type is in fact defined in corelib. This means we can
```

Comment:
Describing why it is ok to move and rename the type should be in the change description. It is not useful as a permanent comment. The fact that the names got changed at some point in the past is not important for somebody understanding this code in future.

Reply:
Right, yeah that's a very good point 😄
My thinking was that it might've been useful to leave some comment to indicate whether the names were important to make the type work, in case someone else wanted to make some more changes in the future, but I agree that it's fine to just indicate this in the commit message and it'll show up in the history anyway (and they'd likely go double check the VM anyway for good measure). Fixed in https://github.com/dotnet/runtime/pull/54246/commits/39ec135957721fe89fbafae1053e9814ad3513f1 and removed the comment to be leave the type declaration as it was originally.